src/Transport/Topology/MetaModel/EntityInfo.cs

```diff
@@ -22,6 +23,16 @@ public EntityInfo()
 
         protected bool Equals(EntityInfo other)
         {
+            if (Type == EntityType.Subscription)
```

Comment:
That and also bringing `EntityInfo` to the level of ASB - I.e. all entities have a path, but not the name. A path for a subscription will be topic path + subscription name. That way, comparing entities` path value will always be correct action.